### PR TITLE
Handle corner cases in LinearLinterpolationScale.v_compute()

### PR DESCRIPTION
**Pull request by mattpap**
fixes https://github.com/holoviz/holoviews/issues/4978

### CODE REVIEW

File: bokehjs/test/unit/models/scales/linear_interpolation_scale.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import {Range1d} from "@bokehjs/models/ranges/range1d"
+
+describe("models/scales/linear_interpolation_scale module", () => {
+
+  describe("LinearInterpolationScale model", () => {
+
+    it("should support vector mapping values", () => {
+      const scale = new LinearInterpolationScale({
+        binning: [0, 1, 2, 300, 39999],
+        source_range: new Range1d({start: 0, end: 40000}),
```
Comment (mattpap) [inline bokehjs/test/unit/models/scales/linear_interpolation_scale.ts:13]: @philippjfr, @jlstevens, so the root cause of the problem was that `binning[-1] != source_range.end` (in the notebook; this test is a simplification of that). Should binning bounds be equal to range bounds? If that's the case, then this PR doesn't fix anything, and it's binning computation that needs to be addressed.

Comment (philippjfr) [reply bokehjs/test/unit/models/scales/linear_interpolation_scale.ts:13]: Hmm, I'd have to dig into that again. If I understand correctly `source_range.end` is the upper bound of the colorbar? I don't think they need to be equal although they usually would be.

Comment (mattpap) [reply bokehjs/test/unit/models/scales/linear_interpolation_scale.ts:13]: > If I understand correctly source_range.end is the upper bound of the colorbar?

Yes.

Comment (mattpap) [reply bokehjs/test/unit/models/scales/linear_interpolation_scale.ts:13]: This PR can be merged regardless, as it makes sure that whatever happens, there won't be out of bounds indexing, but I want to make sure, that if there's a deeper problem with this code, it's actually addressed.

Comment (jlstevens) [reply bokehjs/test/unit/models/scales/linear_interpolation_scale.ts:13]: Agreed: it is worth investigating if there is a deeper issue but I think protecting against out-of-bounds indexing makes sense regardless.
